android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java

```diff
@@ -182,46 +149,22 @@ private void setUpAssetList(NetworkInfo selectedNetwork) {
-                                                        && accountInfo.name.equals(mName)) {
-                                                    AccountInfo[] accountInfos = new AccountInfo[1];
-                                                    accountInfos[0] = accountInfo;
-                                                    WalletListItemModel thisAccountItemModel =
-                                                            new WalletListItemModel(
-                                                                    Utils.getCoinIcon(mCoinType),
-                                                                    mName, mAddress, null, null,
-                                                                    mIsImported);
-                                                    Utils.setUpTransactionList(this, accountInfos,
-                                                            allNetworks, thisAccountItemModel,
-                                                            assetPrices, fullTokenList,
-                                                            nativeAssetsBalances,
-                                                            blockchainTokensBalances,
-                                                            selectedNetwork,
-                                                            walletListItemModelList -> {
-                                                                showTransactionList(
-                                                                        walletListItemModelList);
-                                                            });
-                                                    break;
-                                                }
-                                            }
+        AccountInfo[] accounts = new AccountInfo[1];
+        accounts[0] = mAccountInfo;
+
+        LiveDataUtil.observeOnce(
+                mWalletModel.getCryptoModel().getNetworkModel().mCryptoNetworks, allNetworks -> {
+                    Utils.getTxExtraInfo(new WeakReference<>(this), TokenUtils.TokenType.ALL,
+                            allNetworks, selectedNetwork, accounts, null, false,
+                            (assetPrices, fullTokenList, nativeAssetsBalances,
+                                    blockchainTokensBalances) -> {
+                                WalletListItemModel thisAccountItemModel =
+                                        WalletListItemModel.MakeForAccountInfo(mAccountInfo);
+                                Utils.setUpTransactionList(this, accounts, allNetworks,
+                                        thisAccountItemModel, assetPrices, fullTokenList,
+                                        nativeAssetsBalances, blockchainTokensBalances,
+                                        selectedNetwork, walletListItemModelList -> {
+                                            showTransactionList(walletListItemModelList);
```

Comment:
No need to fetch keyringInfo and search for and account in it as we already have an accountInfo instance